patches/chromium/can_create_window.patch

```diff
@@ -5,68 +5,129 @@ Subject: can_create_window.patch
+index 6097c43bff12bb621a2717bc6f2fe49ff6acb693..6894d5a4e4fdb9a40c06aa04cda796cb1147b937 100644
 --- a/content/public/browser/content_browser_client.h
 +++ b/content/public/browser/content_browser_client.h
-@@ -178,6 +178,7 @@ class RenderFrameHost;
+@@ -177,6 +177,7 @@ class RenderFrameHost;
  class RenderProcessHost;
  class RenderViewHost;
  class ResourceContext;
 +class ResourceRequestBody;
  class SerialDelegate;
  class SiteInstance;
  class SpeechRecognitionManagerDelegate;
-@@ -767,6 +768,8 @@ class CONTENT_EXPORT ContentBrowserClient {
+@@ -811,6 +812,8 @@ class CONTENT_EXPORT ContentBrowserClient {
        const std::string& frame_name,
        WindowOpenDisposition disposition,
        const blink::mojom::WindowFeatures& features,
-+      const std::vector<std::string>& additional_features,
++      const std::string& raw_features,
 +      const scoped_refptr<network::ResourceRequestBody>& body,
        bool user_gesture,
        bool opener_suppressed,
        bool* no_javascript_access);
+diff --git a/content/public/browser/web_contents_delegate.cc b/content/public/browser/web_contents_delegate.cc
+index 7657acfc7d50db1a0064c2fccd4fcc3c4d6a9fff..a11a9efabdb1a04ba9eb2f3c67c8906f7732c54e 100644
+--- a/content/public/browser/web_contents_delegate.cc
++++ b/content/public/browser/web_contents_delegate.cc
+@@ -26,6 +26,16 @@ namespace content {
+
+ WebContentsDelegate::WebContentsDelegate() = default;
+
++void WebContentsDelegate::WebContentsCreated(WebContents* source_contents,
++    int opener_render_process_id,
++    int opener_render_frame_id,
++    const mojom::CreateNewWindowParams& params,
++    WebContents* new_contents) {
++  WebContentsCreated(source_contents, opener_render_process_id,
++                     opener_render_frame_id, params.frame_name,
++                     params.target_url, new_contents);
++}
++
+ WebContents* WebContentsDelegate::OpenURLFromTab(WebContents* source,
+                                                  const OpenURLParams& params) {
+   return nullptr;
+diff --git a/content/public/browser/web_contents_delegate.h b/content/public/browser/web_contents_delegate.h
+index 23c2a2bdaa77d6bf99b59c65d70ac83adc83077a..b5c06f3d43823e0eb6c3e43afc2500868594e0bf 100644
+--- a/content/public/browser/web_contents_delegate.h
++++ b/content/public/browser/web_contents_delegate.h
+@@ -16,6 +16,7 @@
+ #include "base/strings/string16.h"
+ #include "build/build_config.h"
+ #include "content/common/content_export.h"
++#include "content/common/frame.mojom.h"
+ #include "content/public/browser/bluetooth_chooser.h"
+ #include "content/public/browser/bluetooth_scanning_prompt.h"
+ #include "content/public/browser/invalidate_type.h"
+@@ -334,6 +335,12 @@ class CONTENT_EXPORT WebContentsDelegate {
+       const std::string& partition_id,
+       SessionStorageNamespace* session_storage_namespace);
+
++  virtual void WebContentsCreated(WebContents* source_contents,
```

Comment:
I prefer using a different name than `WebContentsCreated` to make code easier to understand, otherwise it is hard distinguish between the 2 methods with the same name.

Reply:
@zcbenz I resolved this, can I have another look?